src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala

```diff
@@ -527,7 +527,8 @@ class SchedulerActions(
   def scale(driver: SchedulerDriver, app: AppDefinition): Unit = {
     import SchedulerActions._
 
-    def launchedNotLost(t: Task) = t.launched.isDefined && t.mesosStatus.fold(false)(_.getState != TaskState.TASK_LOST)
+    def launchedNotLost(t: Task) = t.launched.isDefined && !t.isMesosLost
```

Comment:
this should be `!t.isUnreachable` and the def should be named accordingly. Every terminal task will be expunged from state, so `Gone, Unknown, Lost` will never apply here. The background is that we only want to consider launched _reachable_ tasks here, so it makes sense to evaluate Unreachable only.

That said, you could add an implicit `isReachable` (Staging, Started, Running, Killing), and boil this down to
```
val launchedCount = taskTracker.countAppTasksSync(app.id, _.isReachable)
``` 
without the need for the extra def